libs/sdk-ui-tests-e2e/cypress/tools/layoutRow.ts

```diff
@@ -29,4 +32,64 @@ export class LayoutRow {
 
         return this;
     }
+
+    getHeader() {
+        return new SectionHeader(this.rowIndex);
+    }
+
+    addAbove(name: InsightTitle) {
+        const catalog = new InsightsCatalog().searchExistingInsight(name);
+        this.dragAndDropItems(catalog.getInsightSelector(name), `${this.getSelector()} .row-hotspot`);
+        catalog.clearSearch();
+        return this;
+    }
+
+    addLast(name: InsightTitle) {
+        const catalog = new InsightsCatalog().searchExistingInsight(name);
+        this.dragAndDropItems(catalog.getInsightSelector(name), `.s-last-drop-position`);
+        catalog.clearSearch();
+        return this;
+    }
+
+    addInsightPlaceholder() {
+        this.dragAndDropItems(".s-add-insight", ".s-drag-info-placeholder-drop-target");
+        return this;
+    }
+
+    addKpiPlaceholder() {
+        this.dragAndDropItems(".s-add-kpi:not(.disabled)", ".s-drag-info-placeholder-drop-target");
+        return this;
+    }
+
+    hasCountOfHeaderSection(count: number) {
+        cy.get(".s-fluid-layout-row-header").should("have.length", count);
+        return this;
+    }
+
+    dragAndDropItems(source: string, target: string) {
+        cy.get(source).should("exist");
+        cy.get(source).parent().trigger("dragstart", { force: true });
+        cy.get(target).should("be.visible");
+        cy.wait(1000);
+        cy.get(target).trigger("drop");
+        cy.wait(1000);
```

Comment:
why do we need cy.wait

Reply:
check dragItems(source: string, target: string) also

Reply:
need wait here so that it works more stability. Sometimes, D&D too fast, the widget hasn't dropped to target position

Reply:
please consider and reduce the waiting time, 1s is a long time, and these actions drag and drop widget will be use many many times in our test suites. please check again, try to delete wait and run again, if it still fail then I think around 300ms is enough

Reply:
check other places